libraries/AP_Soaring/SoaringController.cpp

```diff
@@ -0,0 +1,407 @@
+#include "SoaringController.h"
+
+extern const AP_HAL::HAL& hal;
+
+
+// ArduSoar parameters
+const AP_Param::GroupInfo SoaringController::var_info[] = {
+    // @Param: ACTIVE
+    // @DisplayName: Is the soaring mode active or not
+    // @Description: Toggles the soaring mode on and off
+    // @Units: boolean
+    // @Range: 0 1
+    // @User: Advanced
+    AP_GROUPINFO("ACTIVE", 1, SoaringController, soar_active, 0),
```

Comment:
The intent here is that the user wouldn't switch this mode in flight. In other words, the user commits to the soaring mode for the entire mission duration. I might be seeing what you are driving at when asking this question -- whether it's possible to engage and disengage soaring mode opportunistically -- but this has not been tested so far. For the duration of the flight, you should be either in the usual, powered mode, or in the soaring mode, which can engage the throttle when necessary but generally tries to keep the plane in the air without using power.
The intent here is that the user wouldn't switch this mode in flight. In other words, the user commits to the soaring mode for the entire mission duration. I might be seeing what you are driving at when asking this question -- whether it's possible to engage and disengage soaring mode opportunistically -- but this has not been tested so far. For the duration of the flight, you should be either in the usual, powered mode, or in the soaring mode, which can engage the throttle when necessary but generally tries to keep the plane in the air without using power. ACTIVE sounds like the right name for this parameter to me.